data_tamer_cpp/include/data_tamer/logged_value.hpp

```diff
@@ -49,15 +49,22 @@ class LoggedValue
   [[nodiscard]] T get();
 
   /// This method allows us to access a reference to the object in a thread-safe way.
-  /// A mutex remain locked as long as LockedRef<> exist, therefore you should destroy
+  /// A mutex remain locked as long as LockedPtr<> exists, therefore you should destroy
   /// it as soon as you used the reference. Example:
   ///
-  /// if(auto ref = value->getMutablePtr())
+  /// if(auto ref = value->getLockedPtr())
   /// {
   ///   *ref += 1;
   /// }
   [[nodiscard]] LockedPtr<T> getLockedPtr();
 
+  /**
+   * @brief This method is the same as getLockedPtr but does not block
+   *
+   * @return LockedPtr<T>
+   */
+  [[nodiscard]] LockedPtr<T> tryGetLockedPtr();
```

Comment:
I have a problem withg this.

Conceptually, `tryLock` should mean "sorry, I was not able to lock it, and I am telling you that it is NOT locked".

Instead here it is more like "you will always get a result, doesn't matter if it is unsafe or not".

What you implemented here is `getUnsafePtr()`

Reply:
Should this return a `tl::expected` or something, instead? I'm also fine to just close this PR and upgrade to your `ProtectedValue` that you linked, if you think that would be better. We could also just remove the `tryGetLockedPtr` and make this PR just add a test and some documentation.

Reply:
I have removed this entirely, as well as the changes to the LockedPtr constructor.